modules/currency-l0/src/main/scala/io/constellationnetwork/currency/l0/snapshot/programs/Download.scala

```diff
@@ -71,6 +72,18 @@ object Download {
 
       nodeStorage
         .tryModifyState(NodeState.WaitingForDownload, NodeState.DownloadInProgress, NodeState.WaitingForObserving)(start)
+        .flatTap { downloadResult =>
+          val (incrementalSnapshot, _) = downloadResult
+          logger
+            .info(s"[Download] Cleanup for snapshots greater than ${incrementalSnapshot.ordinal}")
+            .handleErrorWith { err =>
+              logger.error(err)("Error during cleanup snapshots on Download") >>
+                Async[F].raiseError(new IllegalStateException(s"Error during cleanup snapshots on Download"))
+            }
```

Comment:
sorry, I meant to add the error handling to the `currencySnapshotCleanupStorage.cleanupAbove` invocation. There is no reason to add the error handling here as the `logger.info` call should not fail.

Reply:
Looking more closely at the `cleanupAbove` method though, you are logging the error already though so maybe that handling isn't needed here.